modules/k8s/namespace.go

```diff
@@ -17,20 +17,40 @@ func CreateNamespace(t testing.TestingT, options *KubectlOptions, namespaceName
 
 // CreateNamespaceE will create a new Kubernetes namespace on the cluster targeted by the provided options.
 func CreateNamespaceE(t testing.TestingT, options *KubectlOptions, namespaceName string) error {
+	namespaceObject := metav1.ObjectMeta{
+		Name: namespaceName,
+	}
+	return CreateNamespaceWithMetadataE(t, options, namespaceObject)
+}
+
+// CreateNamespaceWithMetadataE will create a new Kubernetes namespace on the cluster targeted by the provided options and
+// with the provided metadata. This method expects the entire namespace ObjectMeta to be passed in, so you'll need to set the name within the ObjectMeta struct yourself.
+func CreateNamespaceWithMetadataE(t testing.TestingT, options *KubectlOptions, namespaceObjectMeta metav1.ObjectMeta) error {
 	clientset, err := GetKubernetesClientFromOptionsE(t, options)
 	if err != nil {
 		return err
 	}
 
 	namespace := corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: namespaceName,
-		},
+		ObjectMeta: namespaceObjectMeta,
 	}
 	_, err = clientset.CoreV1().Namespaces().Create(context.Background(), &namespace, metav1.CreateOptions{})
 	return err
 }
 
+// CreateNamespaceWithMetadata will create a new Kubernetes namespace on the cluster targeted by the provided options and
+// with the provided metadata. This method expects the entire namespace ObjectMeta to be passed in, so you'll need to set the name within the ObjectMeta struct yourself.
+func CreateNamespaceWithMetadata(t testing.TestingT, options *KubectlOptions, namespaceObjectMeta metav1.ObjectMeta) {
```

Comment:
This method should call `CreateNamespaceWithMetadataE`, and call `require.NoError` on the error returned, rather than duplicating all the code.

Reply:
ah yeah, sorry about that. Fixed